File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.cpp
class Solution {
   public:
    int countPartitions(vector<int>& nums) {
        int n = nums.size();
        int r = accumulate(nums.begin(), nums.end(), 0);
        int l = 0;

        int ans = 0;
        for (int i = 0; i < n - 1; i++) {
            l += nums[i];
            r -= nums[i];
            if ((l - r) % 2 == 0) ans++;
        }

        return ans;
    }
};

File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.rs
impl Solution {
    pub fn count_partitions(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut l: i32 = 0;
        let mut r: i32 = nums.iter().sum();
        let mut ans = 0;
        for i in 0..n - 1 {
            l += nums[i];
            r -= nums[i];
            if (l - r).abs() % 2 == 0 {
                ans += 1;
            }
        }

        ans
    }
}